File: kotlin-pr/50/kotlin/code-quality-performance/ItemProcessor.kt
class ItemProcessor {
    fun processNumbers(count: Int): List<Int> {
        // {fact rule=code-quality-performance@v1.0 defects=1}
        val numbers = mutableListOf<Int>()
        
        for (i in 0 until count) {
            numbers.add(i * 2)
        }
        
        return numbers
        // {/fact}
    }
}

File: c-pr/45/c/code-quality-performance/item_processor.c
#include <stdio.h>
#include <stdlib.h>

int* process_numbers(int count) {
    // {fact rule=code-quality-performance@v1.0 defects=1}
    int* numbers = malloc(sizeof(int));
    int capacity = 1;
    
    for (int i = 0; i < count; i++) {
        if (i >= capacity) {
            capacity *= 2;
            numbers = realloc(numbers, capacity * sizeof(int));
        }
        numbers[i] = i * 2;
    }
    
    return numbers;
    // {/fact}
}

int main() {
    int* numbers = process_numbers(1000);
    printf("First few numbers: %d %d %d\n", numbers[0], numbers[1], numbers[2]);
    free(numbers);
    return 0;
}